mobilecoind-dev-faucet/src/bin/main.rs

```diff
@@ -0,0 +1,289 @@
+    pub inflight_tx_state: Mutex<Option<SubmitTxResponse>>,
+}
+
+impl State {
+    /// Create a new state from config and a logger
+    fn new(config: &Config, logger: &Logger) -> Result<State, String> {
+        // Search for keyfile and load it
+        let account_key = read_keyfile(config.keyfile.clone()).expect("Could not load keyfile");
+
+        // Set up the gRPC connection to the mobilecoind client
+        let grpc_env = Arc::new(grpcio::EnvBuilder::new().cq_count(1).build());
+        let ch = ChannelBuilder::new(grpc_env.clone())
+            .max_receive_message_len(std::i32::MAX)
+            .max_send_message_len(std::i32::MAX)
+            .connect_to_uri(&config.mobilecoind_uri, &logger);
+
+        let mobilecoind_api_client = MobilecoindApiClient::new(ch);
+
+        // Create a monitor using our account key
+        let monitor_id = {
+            let mut req = mc_mobilecoind_api::AddMonitorRequest::new();
+            req.set_account_key((&account_key).into());
+            req.set_num_subaddresses(2);
+            req.set_name("faucet".to_string());
+
+            let resp = mobilecoind_api_client
+                .add_monitor(&req)
+                .map_err(|err| format!("Failed adding a monitor: {}", err))?;
+
+            resp.monitor_id
+        };
+
+        // Get the b58 public address for monitor
+        let monitor_b58_address = {
+            let mut req = mc_mobilecoind_api::GetPublicAddressRequest::new();
+            req.set_monitor_id(monitor_id.clone());
+
+            let resp = mobilecoind_api_client
+                .get_public_address(&req)
+                .map_err(|err| format!("Failed getting public address: {}", err))?;
+
+            resp.b58_code
+        };
+
+        // Get the network minimum fees and compute faucet amounts
+        let faucet_amounts = {
+            let mut result = HashMap::<TokenId, u64>::default();
+
+            let resp = mobilecoind_api_client
+                .get_network_status(&Default::default())
+                .map_err(|err| format!("Failed getting network status: {}", err))?;
+
+            for (k, v) in resp.minimum_fees.iter() {
+                result.insert(k.into(), *v);
+            }
+
+            result
+        };
+
+        let inflight_tx_state = Mutex::new(Default::default());
+
+        Ok(State {
+            mobilecoind_api_client,
+            monitor_id,
+            monitor_b58_address,
+            faucet_amounts,
+            grpc_env,
+            inflight_tx_state,
+        })
+    }
+
+    fn lock_and_check_inflight_tx_state(
+        &self,
+    ) -> Result<MutexGuard<Option<SubmitTxResponse>>, String> {
+        let guard = self.inflight_tx_state.lock().expect("mutex poisoned");
+        if let Some(prev_tx) = guard.as_mut() {
+            let mut tries = 10;
+            loop {
+                let resp = self
+                    .mobilecoind_api_client
+                    .get_network_status(&prev_tx)
```

Comment:
I don't think this is the right call...

Reply:
i'm trying to base it on the other PR, so in the `last-block-info` PR as it currently exists, I think this would be the right call. but if we redo that then we might have to change this

Reply:
oh i'm sorry, this is the wrong call, you're right, I'm supposed to call the get tx status thing. whoops